unix/syscall_illumos.go

```diff
@@ -176,3 +178,55 @@ func (s *Strioctl) SetInt(i int) {
 func IoctlSetStrioctlRetInt(fd int, req uint, s *Strioctl) (int, error) {
 	return ioctlRet(fd, req, uintptr(unsafe.Pointer(s)))
 }
+
+// Event Ports
+
+//sys	port_create() (n int, err error)
+
+func PortCreate() (int, error) {
+	return port_create()
+}
+
+//sys	port_associate(port int, source int, object uintptr, events int, user unsafe.Pointer) (n int, err error)
+
+//TODO
+//func PortAssociateFd(port int, fd int, events int, user unsafe.Pointer) (n int, err error)
+
+func PortAssociateFileObj(port int, f *FileObj, events int, user unsafe.Pointer) (int, error) {
```

Comment:
Blind guess double pointer?

Reply:
I can't for the life of me remember how pointer conversion from pointer to uintptr works. Maybe unsafe.Pointer is the wrong converter.